src/arduino.cc/arduino-builder/main.go

```diff
@@ -249,8 +248,7 @@ func main() {
 		_, err := os.Stat(buildPath)
 		if err != nil {
 			fmt.Fprintln(os.Stderr, err)
-			defer os.Exit(1)
-			return
+			os.Exit(1)
```

Comment:
Yup that's another way too, BTW my preference is to handle everything with golang's `error` until the very end and use the `main` function to convert it into an error code. BTW I guess we can work this out after merging this PR :-)